aws/src/test/java/org/apache/iceberg/aws/TestAwsClientFactories.java

```diff
@@ -132,6 +139,119 @@ public void testLakeFormationAwsClientFactorySerializable() throws IOException {
+          .hasToString("https://test-bucket.s3.amazonaws.com/test-key");
+    }
+  }
+
+  @Test
+  public void
+      testDefaultAwsClientFactoryWithCredentialsProviderAndCreateMethodWithCustomProperties() {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(AwsProperties.CLIENT_REGION, Region.AWS_GLOBAL.toString());
+    properties.put(
+        AwsProperties.CLIENT_CREDENTIALS_PROVIDER, CustomCredentialsProvider.class.getName());
+    properties.put(AwsProperties.CLIENT_CREDENTIALS_PROVIDER + ".param1", "value1");
+
+    AwsClientFactory defaultAwsClientFactory = AwsClientFactories.from(properties);
+    assertAwsClientFactory(defaultAwsClientFactory);
+  }
+
+  @Test
+  public void testDefaultAwsClientFactoryWithInvalidCredentialsProvider() {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(AwsProperties.CLIENT_REGION, Region.AWS_GLOBAL.toString());
+    properties.put(AwsProperties.CLIENT_CREDENTIALS_PROVIDER, "invalidClassName");
+    Assertions.assertThatThrownBy(() -> AwsClientFactories.from(properties).s3().close())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining(
+            "Cannot load class invalidClassName, it does not exist in the classpath");
+  }
+
+  @Test
+  public void testDefaultAwsClientFactoryWithInvalidCredentialsProviderWithValidClass() {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(AwsProperties.CLIENT_REGION, Region.AWS_GLOBAL.toString());
+    properties.put(
+        AwsProperties.CLIENT_CREDENTIALS_PROVIDER,
+        InvalidNoInterfaceDynamicallyLoadedCredentialsProvider.class.getName());
+    Assertions.assertThatThrownBy(() -> AwsClientFactories.from(properties).s3().close())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining(
+            "Cannot initialize org.apache.iceberg.aws.TestAwsClientFactories$InvalidNoInterfaceDynamicallyLoadedCredentialsProvider, it does not implement software.amazon.awssdk.auth.credentials.AwsCredentialsProvider");
+  }
+
+  @Test
+  public void testDefaultAwsClientFactoryCredentialsProviderWithNoCreateMethod() {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(AwsProperties.CLIENT_REGION, Region.AWS_GLOBAL.toString());
+    properties.put(
+        AwsProperties.CLIENT_CREDENTIALS_PROVIDER,
+        CustomCredentialsProviderWithNoCreateMethod.class.getName());
+    Assertions.assertThatThrownBy(() -> AwsClientFactories.from(properties).s3().close())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining(
+            "Cannot create an instance of org.apache.iceberg.aws.TestAwsClientFactories$CustomCredentialsProviderWithNoCreateMethod, it does not contain a static 'create' or 'create(Map<String, String>)' method");
+  }
+
+  // publicly visible for testing to be dynamically loaded
+  public static class InvalidNoInterfaceDynamicallyLoadedCredentialsProvider {
```

Comment:
I would probably call this public static class NoInterface { }, although you probably don't need a custom class here since you can just pass a java.lang.Object if you wanted. Either way I'd try to shorten the name a bit so that the test is a bit easier to read.

Reply:
Renamed all test provider class names with shorter name